### PR TITLE
Miscalenous small cleanups/additions

### PR DESCRIPTION
**Pull request by mwutzke**
* A number of small usability / documentation improvements
* An initial Vagrant machine to isolate simulation from the developers host environment
* Force all scripts to explicitly use python3

### CODE REVIEW

File: tests/integration/mininet_tests.py [hunk 1]
```diff
@@ -3819,7 +3819,7 @@ def test_untagged(self):
                 '10.0.0.0/24 next-hop 10.0.0.254',
                 '10.0.1.0/24 next-hop 10.0.0.1',
                 '10.0.2.0/24 next-hop 10.0.0.2',
-                '10.0.2.0/24 next-hop 10.0.0.2'):
+                '10.0.3.0/24 next-hop 10.0.0.2'):
```
Comment (cglewis) [inline tests/integration/mininet_tests.py:3822]: Should the next hop be 10.0.0.3 here?

Comment (mwutzke) [reply tests/integration/mininet_tests.py:3822]: No I don't believe so, as the config used for this test-case has:
```
            - route:
                ip_dst: "10.0.3.0/24"
                ip_gw: "10.0.0.2"
```

Comment (cglewis) [reply tests/integration/mininet_tests.py:3822]: ah, good catch. I was just following the pattern and made an assumption that it should be 3.

**Comment by anarkiwi** [comment]
Thanks for your contribution! This is quite a "busy" PR. Would you be able to help us by splitting it up into its various themes?

**Comment by mwutzke** [comment]
Sure thing - can help with that. I think there are 2-3 I can split it into.

**Comment by mwutzke** [comment]
Abandoning this uber-PR. Will break up into more discrete PRs

**Pull request closed without merging**